### ISSUE TITLE
Wallpaper starts playing whenever Explorer.exe is active

### ISSUE THREAD

**Issue by skygate2012** [rocksdanister/lively#1630]
**Is your feature request related to a problem? Please describe.**
For example, it starts playing when I click the taskbar buttons, or when Alt+Tab is pressed, or when Start Menu is opened.

**Describe the solution you'd like**
It should only play when the desktop is focused, I don't know if this is a bug or a feature though.

**Describe alternatives you've considered**
I think it can match the window class instead of the process name. The class for desktop is WorkerW.

**Comment by rocksdanister** [comment]
This is by design, there is a class whitelist that ignores pause rules:
https://github.com/rocksdanister/lively/blob/c27d2d04e9d4e921c83ba74465e0869402e4fc83/src/Lively/Lively/Core/Suspend/Playback.cs#L26-L44

Its because taskbar and other Windows processes can become foreground process when user opens them even while on desktop.

**Comment by skygate2012** [comment]
Ah I see. May I suggest a better solution? Compare the window position of the opened windows and the desktop to determine if it is covered. I've figured out the following in AutoHotkey. It also solves the problem of desktop being partially covered by an active window. And of course, you can call this on window event instead of pooling.

```
#Requires AutoHotkey v2.0
#SingleInstance Force
loop
	tooltip "Desktop is " (IsDesktopCovered() ? "fully covered" : "visible or partially visible")
Return

IsDesktopCovered() {
	Static hDesktop, X2, Y2, W2, H2
	if !IsSet(hDesktop) {
		hDesktop := WinGetID("ahk_class WorkerW ahk_exe explorer.exe")
		WinGetPos(&X2, &Y2, &W2, &H2, hDesktop)
		WinGetPos(, , , &H, "ahk_class Shell_TrayWnd ahk_exe explorer.exe")
		H2 -= H
	}
	covered := false
	; Loop through opened windows
	for this_id in WinGetList(,, "Program Manager")
	{
		;Desktop is at the topmost, which means no other window is on top, assume not covered
		if (this_id = hDesktop)
			return false
		if WinExist(this_id) {
			; Ignore the Alt+Tab screen because its actual size covers the whole screen
			if (WinGetClass(this_id) = "MultitaskingViewFrame")
				continue
			WinGetPos(&X1, &Y1, &W1, &H1, this_id)
			; Whether if 1 completely covers 2
			if (covered := X1 <= X2 && Y1 <= Y2 && X1+W1 >= X2+W2 && Y1+H1 >= Y2+H2)
				return true
		}
	}
}

Esc::ExitApp
```

**Comment by rocksdanister** [comment]
Thanks for the suggestion, will take a look when I'm free.

**Comment by rocksdanister** [comment]
Have you tried verifying the algorithm? using command utility you can run the ahk script directly:

https://github.com/rocksdanister/lively/wiki/Command-Line-Controls#wallpaper-playback
https://github.com/rocksdanister/lively/wiki/Command-Line-Controls#autohotkey

Make sure all Lively Wallpaper playback settings are set to Nothing before testing.

**Comment by skygate2012** [comment]
Yes, I tried the following and it works as expected, with some delay because of the huge command line binary.

```
#Requires AutoHotkey v2.0
#SingleInstance Force

prevState := true
Loop
{
	isCovered := IsDesktopCovered()
	if (isCovered != prevState)
		Run "cmd.exe /c livelycu.exe app --play " (isCovered ? "false" : "true"), , "Hide"
	prevState := isCovered
}
```

**Comment by rocksdanister** [comment]
resolved: https://github.com/rocksdanister/lively/commit/034e9a6b4ca40c59caf6a6842667dd3494404f34
